Naughts_and_Crosses.py

```diff
@@ -1,122 +1,237 @@
+        self._board_size()
+        self.board_state = [[' ' for i in range(self.board_size)] for i in range(self.board_size)]
+        self.generate_win_condition()
+        self.generate_empty_spaces()
+    
+    def generate_empty_spaces(self):  #Generate the empty spaces from the board size
+        self.empty_spaces = []
+        for i in range(self.board_size):
+            for j in range(self.board_size):
+                self.empty_spaces.append([i,j])
+
+    def generate_win_condition(self): #Generates the win condition set from the board size
+        self.win_condition = []
+
+        #Vertical Wins
+        for i in range(self.board_size):
+            _sub_win_condition = []
+            for j in range(self.board_size):
+                _sub_win_condition.append([i, j])
+            self.win_condition.append(_sub_win_condition)
+
+        #Horizontal Wins
+        for i in range(self.board_size):
+            _sub_win_condition = []
+            for j in range(self.board_size):
+                _sub_win_condition.append([j, i])
+            self.win_condition.append(_sub_win_condition)
+
+            #Top Left Bottomw Right
+        _sub_win_condition = []
+        for i in range(self.board_size):
+            _sub_win_condition.append([i, i])
+        self.win_condition.append(_sub_win_condition)
+ 
+            #Top Right Bottom Left
+        _sub_win_condition = []
+        for i in range(self.board_size):
+            _sub_win_condition.append([i, (self.board_size - 1) - i])
+        self.win_condition.append(_sub_win_condition)
+
+    def _board_size(self): #Asks the user what size board they want to play on
+        self.board_size = int(input("What board size would you like? (N x N)  "))
+
+    def printboard(self): #Simply prints an empty list into a matrix
+        for item in self.board_state:
+                print('|'.join(item))
+
+    def check_free_spaces(self): #Checks if there are any empty spaces left on the board
+        if self.empty_spaces == []:
+            return False
+        return True
 
 class Game():
-    def start_game(self):
-	    s = input('Would you like to play a game? (y/n)  ')
-        if s == 'n':
-            print('Okay fine then...')
-            start_game(self)
-        elif s == 'y':
-            board_state = board_state_start
-            start_game(self)
+    running = True
+
+    def __init__(self): #Generates the board, then the players, then runs the game
+        self.board = Board()
+        self._generate_players()
+        self.run_game()
+
+    def run_game(self): #Runs the game by printing the board, asking who goes first, then loops the turn function until
+        self.board.printboard()  #the end of the game
+        self.first_turn()
+        while self.running == True:
+            self.turn()
+            self.board.printboard()
+            if self.check_win() == True:
+            	break
+            if self.check_draw() == True:
+            	break
```

Comment:
What's the difference between breaking this loop to go the end_game() func, and setting self.running = False?

Breaks are generally bad, just set self.running = False.

Also you can do: if self.check_win() or self.check_draw():

And another:

`if self.condition == True` === `if self.condition`